exercises/react/example/Reactor.ceylon

```diff
@@ -0,0 +1,88 @@
+import ceylon.collection { HashMap, MutableMap }
+
+alias Element => Integer;
+
+class Reactor() {
+  shared abstract class Cell() {
+    shared formal Element currentValue;
+    variable ComputeCell[] dependents = [];
+    shared void addDependent(ComputeCell c) {
+      dependents = dependents.withTrailing(c);
+    }
+    shared void eachDependent(Anything(ComputeCell) f) {
+      for (dep in dependents) {
+        f(dep);
+      }
+    }
+  }
+
+  shared class InputCell(Element initialValue) extends Cell() {
+    variable Element currentValue_ = initialValue;
+    shared actual Element currentValue => currentValue_;
+    assign currentValue {
+      currentValue_ = currentValue;
+      eachDependent((d) => d.propagate());
+      eachDependent((d) => d.fireCallbacks());
+    }
+  }
+
+  shared InputCell newInputCell(Element initialValue) {
```

Comment:
it's not strictly necessary to delegate the creation to a `Reactor`, but it is consistent with `newComputeCell*` functions, which are necessary because of https://github.com/exercism/xceylon/pull/20#discussion_r96332335.